popdat/src/import_census.rs

```diff
@@ -1,11 +1,95 @@
-use map_model::Map;
-
 use crate::CensusArea;
+use abstutil::Timer;
+use geo::algorithm::intersects::Intersects;
+use geojson::GeoJson;
+use map_model::Map;
+use std::convert::TryFrom;
 
 impl CensusArea {
-    pub fn find_data_for_map(_map: &Map) -> Result<Vec<CensusArea>, String> {
-        // TODO importer/src/utils.rs has a download() helper that we could copy here. (And later
-        // dedupe, after deciding how this crate will integrate with the importer)
-        todo!()
+    pub fn find_data_for_map(map: &Map, timer: &mut Timer) -> Result<Vec<CensusArea>, String> {
+        // TODO eventually it'd be nice to lazily download the info needed. For now we expect a
+        // prepared geojson file to exist in data/system/<city>/population_areas.geojson
+        //
+        // When we implement downloading, importer/src/utils.rs has a download() helper that we
+        // could copy here. (And later dedupe, after deciding how this crate will integrate with
+        // the importer)
+        let path = abstutil::path(format!(
+            "system/{}/population_areas.geojson",
+            map.get_name().city
+        ));
+        let bytes = abstutil::slurp_file(&path)?;
+        debug!("parsing geojson at path: {}", &path);
+
+        // TODO - can we change to Result<_,Box<dyn std::error::Error>> and avoid all these map_err?
+        let str = String::from_utf8(bytes).map_err(|e| e.to_string())?;
+        timer.start("parsing geojson");
+        let geojson = str.parse::<GeoJson>().map_err(|e| e.to_string())?;
+        timer.stop("parsing geojson");
+        let mut results = vec![];
+        let collection =
+            geojson::FeatureCollection::try_from(geojson).map_err(|e| e.to_string())?;
+
+        let map_area = map.get_boundary_polygon();
+        let bounds = map.get_gps_bounds();
+
+        use geo::algorithm::map_coords::MapCoordsInplace;
+        let mut geo_map_area: geo::Polygon<_> = map_area.clone().into();
+        geo_map_area.map_coords_inplace(|c| {
+            let projected = geom::Pt2D::new(c.0, c.1).to_gps(bounds);
+            (projected.x(), projected.y())
+        });
+
+        debug!("collection.features: {}", &collection.features.len());
+        timer.start("converting to `CensusArea`s");
+        for feature in collection.features.into_iter() {
+            let population = feature.property("population");
+            let total_population = match population {
+                Some(serde_json::Value::Number(n)) => n
```

Comment:
geojson prop parsing is depressingly verbose. I hope this gets better soon! https://github.com/georust/geojson/issues/157